src/probnum/filtsmooth/timeseriesposterior.py

```diff
@@ -0,0 +1,190 @@
+"""Abstract Base Class for posteriors over states after applying filtering/smoothing."""
+
+import abc
+from typing import Optional, Union
+
+import numpy as np
+
+from probnum import _randomvariablelist, random_variables
+from probnum.type import (
+    ArrayLikeGetitemArgType,
+    FloatArgType,
+    RandomStateArgType,
+    ShapeArgType,
+)
+
+DenseOutputLocationArgType = Union[FloatArgType, np.ndarray]
+"""TimeSeriesPosteriors and derived classes can be evaluated at a single location 't'
+or an array of locations."""
+
+DenseOutputValueType = Union[
+    random_variables.RandomVariable, _randomvariablelist._RandomVariableList
+]
+"""Dense evaluation of a TimeSeriesPosterior returns a RandomVariable if evaluated at a single location,
+and a _RandomVariableList if evaluated at an array of locations."""
+
+
+class TimeSeriesPosterior(abc.ABC):
+    """Posterior Distribution over States after time-series algorithms such as
```

Comment:
Could you briefly remind me why we need TimeSeriesPosterior, and why just having KalmanPosterior is not sufficient?

Reply:
TimeSeriesPosterior is intended to be a joint interface for all sorts of posterior processes. This includes `KalmanPosterior`, but also `ODESolution` (, and `ParticleFilterPosterior`).

Reply:
I see, thanks!